### PR TITLE
Feature/email only auth

### PR DESCRIPTION
**Pull request by dephillipsmichael**
This PR adds email-only sign up and sign in functionality to ResearchStask.  I have tested this through BP app with email-only and normal email/password and both work as expected.

### CODE REVIEW

File: backbone/src/main/java/org/researchstack/backbone/utils/SerializableUtils.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public static String toBase64String(Serializable object) {
+        String encoded = null;
+        try {
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+            objectOutputStream.writeObject(object);
+            objectOutputStream.close();
+            encoded = Base64.encodeToString(byteArrayOutputStream.toByteArray(), 0);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (liujoshua) [inline backbone/src/main/java/org/researchstack/backbone/utils/SerializableUtils.java]: Should these go to logcat?

File: backbone/src/main/java/org/researchstack/backbone/utils/SerializableUtils.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     * @return the serializable object, or null if something went wrong
+     */
+    public static Serializable fromBase64String(String base64String) {
+        byte[] bytes = Base64.decode(base64String,0);
+        Serializable object = null;
+        try {
+            ObjectInputStream objectInputStream = new ObjectInputStream( new ByteArrayInputStream(bytes) );
+            object = (Serializable)objectInputStream.readObject();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (liujoshua) [inline backbone/src/main/java/org/researchstack/backbone/utils/SerializableUtils.java]: Log for these also

**Review by liujoshua — APPROVED** [review]
Minor error logging comments

File: backbone/src/main/java/org/researchstack/backbone/utils/SerializableUtils.java (hunk 1 above)
Comment (dephillipsmichael) [reply backbone/src/main/java/org/researchstack/backbone/utils/SerializableUtils.java]: Yes good idea, was copy/paste code

**Pull request merged**